src/main/java/com/commercetools/project/sync/util/ClientConfigurationUtils.java

```diff
@@ -0,0 +1,100 @@
+
+  /**
+   * Creates a {@link BlockingSphereClient} with a custom {@code timeout} with a custom {@link
+   * TimeUnit}.
+   *
+   * @param clientConfig the client configuration for the client.
+   * @return the instantiated {@link BlockingSphereClient}.
+   */
+  static SphereClient createClient(@Nonnull final SphereClientConfig clientConfig) {
+
+    final HttpClient httpClient = getHttpClient();
+    final SphereAccessTokenSupplier tokenSupplier =
+        SphereAccessTokenSupplier.ofAutoRefresh(clientConfig, httpClient, false);
+    final SphereClient underlying = SphereClient.of(clientConfig, httpClient, tokenSupplier);
+    final SphereClient retryClient = withRetry(underlying);
+    return withLimitedParallelRequests(retryClient);
+  }
+
+  /**
+   * Gets an asynchronous {@link HttpClient} to be used by the {@link BlockingSphereClient}. Client
+   * is created during first invocation and then cached.
+   *
+   * @return {@link HttpClient}
+   */
+  public static HttpClient getHttpClient() {
+    final AsyncHttpClient asyncHttpClient =
+        new DefaultAsyncHttpClient(
+            new DefaultAsyncHttpClientConfig.Builder()
+                .setHandshakeTimeout((int) DEFAULT_TIMEOUT)
+                .build());
+    return AsyncHttpClientAdapter.of(asyncHttpClient);
+  }
+
+  private static SphereClient withRetry(final SphereClient delegate) {
+    final int maxAttempts = 5;
+    final RetryAction scheduledRetry =
+        RetryAction.ofScheduledRetry(
+            maxAttempts, context -> calculateVariableDelay(context.getAttempt()));
+    final RetryPredicate http5xxMatcher =
+        RetryPredicate.ofMatchingStatusCodes(
+            BAD_GATEWAY_502, SERVICE_UNAVAILABLE_503, GATEWAY_TIMEOUT_504);
+    final List<RetryRule> retryRules =
+        Collections.singletonList(RetryRule.of(http5xxMatcher, scheduledRetry));
+    return RetrySphereClientDecorator.of(delegate, retryRules);
+  }
+
+  /**
+   * Computes a variable delay in seconds (grows with attempts count with a random component).
+   *
+   * @param triedAttempts the number of attempts already tried by the client.
+   * @return a computed variable delay in seconds, that grows with the number of attempts with a
+   *     random component.
+   */
+  private static Duration calculateVariableDelay(final long triedAttempts) {
+    final long timeoutInSeconds = TimeUnit.SECONDS.convert(DEFAULT_TIMEOUT, TimeUnit.MILLISECONDS);
+    final long randomNumberInRange = getRandomNumberInRange(50, timeoutInSeconds);
+    final long timeoutMultipliedByTriedAttempts = timeoutInSeconds * triedAttempts;
+    return Duration.ofSeconds(timeoutMultipliedByTriedAttempts + randomNumberInRange);
+  }
+
+  private static long getRandomNumberInRange(final long min, final long max) {
+    return new Random().longs(min, (max + 1)).limit(1).findFirst().getAsLong();
+  }
+
+  private static SphereClient withLimitedParallelRequests(final SphereClient delegate) {
+    final int maxParallelRequests = 20;
```

Comment:
Might be useful to make it configurable.

Reply:
Could be yes. But better create an issue of it. It's a nice-to-have but not so important (not enough time for it now). And would rather have it in a separate PR that has it's tests and reviewed separatley and so on.